lib/fastly/client.rb

```diff
@@ -132,16 +137,20 @@ def build_http_client
 
     def post_and_put(method, path, params = {})
       extras = params.delete(:headers) || {}
+      headers_no_auth = params.delete(:headers_no_auth) || false
       query = make_params(params)
-      resp  = http.send(method, path, query, headers(extras).merge('Content-Type' =>  'application/x-www-form-urlencoded'))
+      resp  = http.send(method, path, query, headers(extras, headers_no_auth).merge('Content-Type' =>  'application/x-www-form-urlencoded'))
       fail Error, resp.body unless resp.kind_of?(Net::HTTPSuccess)
       JSON.parse(resp.body)
     end
 
-    def headers(extras={})
-      headers = fully_authed? ? { 'Cookie' => cookie } : {}
-      headers['Fastly-Key'] = api_key if api_key
-
+    def headers(extras={}, headers_no_auth=false)
```

Comment:
I know this is a bit a of a nit pick, but after discussing this PR internally we agreed that the double-negative makes this less desirable. Would you mind terribly changing this to an affirmative keyword argument? As an example...

```suggestion
    def headers(extras={}, include_auth: true)
```